R/transform_range.R

```diff
@@ -0,0 +1,114 @@
+#' Transform Range
+#'
+#' Transforms results from the source range to the target range. For example,
+#' for transforming source values 1, 2, 3, 4, 5 to 0, 25, 50, 75, 100.
+#'
+#' @param source A vector of values to be transformed
+#'
+#'   A numeric vector is expected.
+#'
+#' @param source_range The permitted source range
+#'
+#'   A numeric vector containing two elements is expected, representing the
+#'   lower and upper bounds of the permitted source range.
+#'
+#' @param target_range The target range
+#'
+#'   A numeric vector containing two elements is expected, representing the
+#'   lower and upper bounds of the target range.
+#'
+#' @param flip_direction Flip direction of the range?
+#'
+#'   The transformed values will be reversed within the target range, e.g.
+#'   within the range 0 to 100, 25 would be reversed to 75.
+#'
+#'   *Permitted Values*: `TRUE`, `FALSE`
+#'
+#' @param outside_range Handling of values outside the source range
+#'
+#'   Values outside the source range (`source_range`) are transformed to `NA`.
+#'
+#'   If `"warning"` or `"error"` is specified, a warning or error is issued if
+#'   `source` includes any values outside the source range.
+#'
+#'   *Permitted Values*: `"NA"`, `"warning"`, `"error"`
```

Comment:
Is this set to "NA" or `NA_character` ?

Reply:
The `outside_range` argument expects `"NA"`.

Reply:
SOrry!!!  So the argument `outside_range = "NA"` will produce NA values?

Reply:
Yes, values outside the source range result in `NA`. If `"warning"` or `"error"` is specified, a warning or error is issued in addition.